code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -154,8 +157,83 @@
 
 	modes = list(MODE_NONE = MODE_SHUTTLE, MODE_SHUTTLE = MODE_NONE)
 
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging
+	name = "atmospheric thermal imaging goggles"
+	desc = "Goggles used by Atmospheric Technician to see the thermal energy of gasses in open areas"
+	icon_state = "trayson-radiation"
+	inhand_icon_state = "trayson-meson"
+	glass_colour_type = /datum/client_colour/glass_colour/gray
+
+	modes = list(MODE_NONE = MODE_ATMOS_THERMAL, MODE_ATMOS_THERMAL = MODE_NONE)
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/toggle_mode(mob/user, voluntary)
+	mode = modes[mode]
+	to_chat(user, "<span class='[voluntary ? "notice":"warning"]'>[voluntary ? "You turn the goggles":"The goggles turn"] [mode ? "to [mode] mode":"off"][voluntary ? ".":"!"]</span>")
+	switch(mode)
+		if(MODE_ATMOS_THERMAL)
+			change_glass_color(user, /datum/client_colour/glass_colour/lightorange)
+		if(MODE_NONE)
+			change_glass_color(user, initial(glass_colour_type))
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.glasses == src)
+			H.update_sight()
+
+	update_appearance()
+	update_action_buttons()
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/update_icon_state()
+	icon_state = inhand_icon_state = "trayson-[mode]"
+	return ..()
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/process()
+	if(!ishuman(loc))
+		return
+	var/mob/living/carbon/human/user = loc
+	if(user.glasses != src || !user.client)
+		return
+	if(mode == MODE_ATMOS_THERMAL)
+		atmos_thermal(user)
+
+#define TEMP_SHADE_CYAN 273.15
+#define TEMP_SHADE_GREEN 283.15
+#define TEMP_SHADE_YELLOW 300
+#define TEMP_SHADE_RED 500
+
+/proc/atmos_thermal(mob/viewer, range = 4, duration = 10)
+	if(!ismob(viewer) || !viewer.client)
+		return
+	for(var/turf/open in view(range, viewer))
+		if(open.blocks_air)
+			continue
+		var/datum/gas_mixture/environment = open.return_air()
+		var/temp = round(environment.temperature)
+		var/image/pic = image('icons/turf/overlays.dmi', open, "greyOverlay", ABOVE_ALL_MOB_LAYER)
+		// Lower than TEMP_SHADE_CYAN should be deep blue
+		switch(temp)
+			if(TCMB to TEMP_SHADE_CYAN)
+				pic.color = COLOR_STRONG_BLUE
+			// Between TEMP_SHADE_CYAN and TEMP_SHADE_GREEN
+			if(274.15 to TEMP_SHADE_GREEN)
+				pic.color = BlendRGB(COLOR_DARK_CYAN, COLOR_LIME, max(round((temp - TEMP_SHADE_CYAN)/(TEMP_SHADE_GREEN - TEMP_SHADE_CYAN), 0.01), 0))
+			// Between TEMP_SHADE_GREEN and TEMP_SHADE_YELLOW
+			if(284.15 to TEMP_SHADE_YELLOW)
+				pic.color = BlendRGB(COLOR_LIME, COLOR_YELLOW, clamp(round((temp-TEMP_SHADE_GREEN)/(TEMP_SHADE_YELLOW - TEMP_SHADE_GREEN), 0.01), 0, 1))
+			// Between TEMP_SHADE_YELLOW and TEMP_SHADE_RED
+			if(301 to TEMP_SHADE_RED)
+				pic.color = BlendRGB(COLOR_YELLOW, COLOR_RED, clamp(round((temp-TEMP_SHADE_YELLOW)/(TEMP_SHADE_RED - TEMP_SHADE_YELLOW), 0.01), 0, 1))
+			// Over TEMP_SHADE_RED should be red
+			if(501 to FUSION_MAXIMUM_TEMPERATURE)
```

Comment:
```suggestion
			if(TEMP_SHADE_RED to INFINITY)
```

Reply:
The number is there so that the condition doesn't overlap

Reply:
It doesn't matter if they overlap, switch handles that itself